### ISSUE TITLE
Side freeze on load with plus plus wikis

### ISSUE THREAD

**Issue by ffiesta** [Zavy86/WikiDocs#122]
### Describe your issue

Slow open sidebar on computer and on android slow open and stay opened on select. My wiki has 293 folders and 729 files.

### Device and settings

Android

### Steps to reproduce

Plus plus wikis 293 folders and 729 files. I need to comment from line 74 to 90 on template.inc.php to don't show 2 level.
![Untitled 2](https://github.com/Zavy86/WikiDocs/assets/25620658/bd1d5c41-5782-4d28-9e80-738044f3754b)
![Untitled 1](https://github.com/Zavy86/WikiDocs/assets/25620658/36d82068-9768-4451-8d90-c8df4b154856)

### Extra fields

- [ ] I'd like to work on this issue

**Comment by Zavy86** [comment]
we can introduce an option to choose how many levels show in the sidebar..
what do you think?

**Comment by ffiesta** [comment]
> we can introduce an option to choose how many levels show in the sidebar.. what do you think?

I tink its good idea in begining, but the possibility in future choose manual option.

**Comment by ffiesta** [comment]
Please see: 
 https://github.com/Zavy86/WikiDocs/pull/142

**Comment by leomoon** [comment]
can we have the page content that freezes on?

we need to replicate it to see what's going on.

**Comment by leomoon** [comment]
I wrote this to create mass amount of documents;
```bash
#!/bin/bash

# Default values
num_categories=15
num_subcategories=7
num_documents=10

# Parse command-line arguments
while getopts "c:s:d:" opt; do
    case ${opt} in
        c )
            num_categories=$OPTARG
            ;;
        s )
            num_subcategories=$OPTARG
            ;;
        d )
            num_documents=$OPTARG
            ;;
        \? )
            echo "Usage: cmd [-c] num_categories [-s] num_subcategories [-d] num_documents"
            exit 1
            ;;
    esac
done

for c in $(seq 1 $num_categories); do
    category=category-$c-$(tr -dc a-z </dev/urandom | head -c 7 ; echo '')
    mkdir ${category}
    for s in $(seq 1 $num_subcategories); do
        subcat=subcat-${c}-${s}-$(tr -dc a-z </dev/urandom | head -c 7 ; echo '')
        mkdir ${category}/${subcat}
        for i in $(seq 1 $num_documents); do
            name=sample-$(tr -dc a-z </dev/urandom | head -c 13 ; echo '')
            title="${name//-/ }"
            title=($title)
            title="${title[@]^}"
            mkdir ${category}/${subcat}/$name
            echo "# $title

This is doc \`$title\`.

## Formatting:

\`\`\`
# sample code $title
Dolor ad excepteur mollit irure qui cillum cillum sunt. Occaecat laboris voluptate dolore eu adipisicing duis sunt dolore laboris aute et do veniam in. Eiusmod est ad minim elit amet esse velit sint proident fugiat.

Deserunt eu deserunt eiusmod velit Lorem anim non do culpa deserunt. Quis aute incididunt voluptate esse consectetur commodo duis ut irure aute. Ut consequat aliquip deserunt ex amet eiusmod commodo nisi Lorem sint consequat.
\`\`\`

Excepteur sunt aute cillum duis ex. Ex pariatur Lorem excepteur excepteur duis quis ipsum sit ad excepteur minim mollit. Sit qui qui mollit ad sint dolore laboris tempor ut ipsum qui eu magna quis.

Eu dolor magna eiusmod voluptate quis aliquip consequat enim consequat dolore consequat dolor velit duis. Anim aliqua consequat occaecat occaecat magna commodo irure aute aliquip reprehenderit et deserunt commodo. Est ad cupidatat proident eu magna dolore qui magna cillum ex.

1. ol_item_1
1. ol_item_2
    - ul_inner_1
    - ul_inner_2
1. ol_item_3
1. ol_item_4

Aliqua cupidatat dolore veniam magna amet commodo velit deserunt nisi magna. Et dolore minim ipsum aliqua veniam quis tempor ipsum dolore ea enim exercitation. Ut commodo incididunt aliqua ullamco anim commodo consectetur ipsum. Incididunt deserunt ullamco pariatur est proident veniam id nulla ad laboris labore eiusmod consectetur aliquip. Nisi consectetur nisi aliqua consequat ullamco consectetur cupidatat velit fugiat. Nisi voluptate proident et duis do esse mollit non.

> Veniam ea proident ea anim deserunt do tempor est tempor consectetur magna sit eiusmod tempor. Laborum veniam consequat labore incididunt. Labore nisi eu est nulla cillum laborum est proident quis sit fugiat in culpa id. Eu tempor eu sunt elit veniam amet tempor elit.

Eiusmod consectetur est aliquip enim adipisicing. Nulla sit consectetur consectetur laboris esse. Ex ut dolore deserunt culpa non aliquip mollit Lorem elit quis consectetur elit est. Anim anim velit ut voluptate sint adipisicing aliqua tempor amet cillum aliqua elit.

- ul_item_1
- ul_item_2
    1. ol_inner_1
    1. ol_inner_2
- ul_item_3
- ul_item_4

Sint consequat amet aliqua nulla. Sint non ex aliquip magna non esse. Consectetur pariatur amet ut elit aliqua nulla. Id incididunt et labore fugiat sint cupidatat duis officia do cupidatat. Dolor minim consequat qui et dolor nisi nulla est esse quis sunt eu elit consequat. Cupidatat duis sunt fugiat pariatur aliquip incididunt non duis non aliqua. Qui ad adipisicing do duis pariatur do qui voluptate.

Lorem non nulla ea aliqua sunt. Ea mollit occaecat elit est occaecat sint labore exercitation officia. Do occaecat anim irure ad sint voluptate tempor elit fugiat veniam. Dolor aliqua sunt ad culpa elit id.

" > ${category}/${subcat}/${name}/content.md
        done
    done
done
```

Save it as `doc-gen`, then `chmod +x doc-gen` and run it like `./projects/scripts/linux/doc-gen -c 7 -s 7 -d 15` which would create 7 categories, 7 subcategories and 7 documents for each... a total of 735 files and 791 folders.

**Comment by ffiesta** [comment]
Yes, see zip attach,

In my previous analysis is that the size screen brake words limitation.  

![Screenshot_20240709-171823_File Manager +](https://github.com/Zavy86/WikiDocs/assets/25620658/e35f2469-5efc-4219-ac13-787d69588163)


I detect that the wikidocs sometimes won't brake text/url, based on screen size, like if the brake as built-in brake text if as xx caracteres.

I tested in smartphone, tablet and sizing browser window.

[datasets-soft-cpanel-tutoriais.zip](https://github.com/user-attachments/files/16146585/datasets-soft-cpanel-tutoriais.zip)

**Comment by leomoon** [comment]
with my script I created 985 docs and 791 folders. Things are still fast and no freeze:

https://github.com/Zavy86/WikiDocs/assets/3268946/31691d3d-4b16-413f-8721-0cda7793c3c3

Could it be your hosting is hitting a limit?

**Comment by ffiesta** [comment]
Hi @Zavy86 @leomoon because some of my wiki are in some way a reference copy from web, I can change my view password and privately share with you to better text this in loko.

If we think it's better please ask.

**Comment by leomoon** [comment]
It is the doc you provided (Tutorials). I'm getting the side menu freeze with that doc.

**Comment by ffiesta** [comment]
> with my script I created 985 docs and 791 folders. Things are still fast and no freeze:
> 
>  many-docs-demo.mp4 
> Could it be your hosting is hitting a limit?

Nope, my ptisp.pt cpanel hosting as 30GB. This happened only with folder files that as long url, as example mobile in UP  after 30 caracteres, and in SIDE after 60 caracteres and chrome/desktop after 300 caracteres.

Tested on brave, chrome edge, and browser that can simulate smartphone screen sizes.

**Comment by leomoon** [comment]
I know what's happening.

- The links are TOO long and they don't break.
- Css thinks this is a desktop but also run mobile-fix.

**Comment by leomoon** [comment]
I have a fix.

**Comment by ffiesta** [comment]
I remember that in past wiki version this happen in **code** from the image, but I think @leomoon solved on 1.0.25 version, with copy button.

![image](https://github.com/Zavy86/WikiDocs/assets/25620658/5bdc1ac1-f512-4133-aad0-492d0d25563d)


Thanks
